lib/ansible/modules/cloud/ovirt/ovirt_vms.py

```diff
@@ -1712,20 +1712,20 @@ def _get_cluster_mappings(module):
     return clusterMappings
 
 
-def _get_vnic_profile_mappings(module):
+def _get_vnic_profile_mappings(module, dc_name):
     vnicProfileMappings = list()
 
     for vnicProfileMapping in module.params['vnic_profile_mappings']:
-        vnicProfileMappings.append(
-            otypes.VnicProfileMapping(
-                source_network_name=vnicProfileMapping['source_network_name'],
-                source_network_profile_name=vnicProfileMapping['source_profile_name'],
-                target_vnic_profile=otypes.VnicProfile(
-                    id=vnicProfileMapping['target_profile_id'],
-                ) if vnicProfileMapping['target_profile_id'] else None,
+        if dc_name == vnicProfileMapping['target_dc_name']:
```

Comment:
Wouldn't it be better to raise some execption if user provide wrong `target_dc_name`, and not silently ignore it?

Reply:
There could be a scenario where the user will prefer to use the default vnic profile the DC provides when the VM is being registered.
In this case, the mapping of ansible might not include the network for the specific DC the VM is being registered to, but we will still want that the VM will be registered.
Besides that, I now see that there is a bug here, since the dc_name is an optional field, so I need to add a check first it it in now none.
I will fix that.